examples/bert.ipynb

```diff
@@ -0,0 +1,274 @@
+        "\n",
+        "\n",
+        "def prepare_sentence_tokens(hf_model: str, sentence: str):\n",
+        "    tokenizer = AutoTokenizer.from_pretrained(hf_model)\n",
+        "    return torch.tensor([tokenizer.encode(sentence)])\n",
+        "\n",
+        "\n",
+        "class OnlyLogitsHuggingFaceModel(torch.nn.Module):\n",
+        "    \"\"\"Wrapper that returns only the logits from a HuggingFace model.\"\"\"\n",
+        "\n",
+        "    def __init__(self, model_name: str):\n",
+        "        super().__init__()\n",
+        "        self.model = AutoModelForSequenceClassification.from_pretrained(\n",
+        "            model_name,  # The pretrained model name.\n",
+        "            # The number of output labels--2 for binary classification.\n",
+        "            num_labels=2,\n",
+        "            # Whether the model returns attentions weights.\n",
+        "            output_attentions=False,\n",
+        "            # Whether the model returns all hidden-states.\n",
+        "            output_hidden_states=False,\n",
+        "            torchscript=True,\n",
+        "        )\n",
+        "        self.model.eval()\n",
+        "\n",
+        "    def forward(self, input):\n",
+        "        # Return only the logits.\n",
+        "        return self.model(input)[0]\n",
+        "\n",
+        "\n",
+        "# Suppress warnings\n",
+        "import warnings\n",
+        "warnings.simplefilter(\"ignore\")\n",
+        "import os\n",
+        "os.environ[\"TOKENIZERS_PARALLELISM\"] = \"true\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# IREE Compilation\n",
+        "\n",
+        "Now, our PyTorch model can be compiled to MLIR and then to a format IREE is able to load and execute."
+      ],
+      "metadata": {
+        "id": "ENzdaB8zogBj"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# The HuggingFace model name to use\n",
+        "model_name = \"philschmid/MiniLM-L6-H384-uncased-sst2\"\n",
+        "\n",
+        "# The sentence to run the model on\n",
+        "sentence = \"The quick brown fox jumps over the lazy dog.\"\n",
+        "\n",
+        "print(\"Parsing sentence tokens.\")\n",
+        "example_input = prepare_sentence_tokens(model_name, sentence)\n",
+        "\n",
+        "print(\"Instantiating model.\")\n",
+        "model = OnlyLogitsHuggingFaceModel(model_name)\n",
+        "\n",
+        "print(\"Tracing model.\")\n",
+        "traced = torch.jit.trace(model, example_input)\n",
+        "\n",
+        "print(\"Compiling with Torch-MLIR\")\n",
+        "linalg_on_tensors_mlir = torch_mlir.compile(\n",
+        "    traced,\n",
+        "    example_input,\n",
+        "    output_type=torch_mlir.OutputType.LINALG_ON_TENSORS)\n",
+        "\n",
+        "print(\"Compiling with IREE\")\n",
+        "# Backend options:\n",
+        "#\n",
+        "# llvm-cpu - cpu, native code\n",
```

Comment:
I wonder if we can have a constant list in the iree_torch package that has this list, and then file a bug upstream in IREE for there being a Python constant for these and for the mapping to drivers (i.e. [this code](https://github.com/dellis23/iree-torch/blob/54b40621dbf46e7a881059cf960d1fa4ef03cad8/python/iree_torch/__init__.py#L104))


These names have changed multiple times and so having fewer places to update would be nice.

Reply:
Yeah, I thought about that a bit, but I'm not quite sure how it would help in presenting it to the user (in this specific case).  Would we store both the backend name and a description and then iterate through each and print them out?

Reply:
I think that just having them in a variable they can reference would be a good start (not all will work anyway -- e.g. vulkan requires vulkan drivers, cuda requires nvidia gpu, etc.). Or better yet, IREE should have a docs page that just lists all the target backends and drivers and their descriptions. (maybe file an IREE bug?)

Reply:
Agreed re: docs.  I'll file a bug.

Reply:
https://github.com/iree-org/iree/issues/10432